Cleanup connection class

RobertMe opened the pull request:
These changes don't change any behavior, but they do clean up a lot of old/unused and duplicate code in the connection class. So the main purpose is to make it easier to maintain the class.

NOTE: Contains commit of #1 so same warning applies. These changes can be reviewed, but please hold of merging until I've figured out what is the best way to do the rename.

mzanetti commented on libkodimote/kodiconnection.cpp:
```diff
@@ -259,10 +247,24 @@ void KodiConnectionPrivate::slotConnected()
+    }
+    m_connecting = false;
+    emit m_notifier->connectionChanged();
+    m_host->setPersistent(true);
```
I think the host should only be stored when actually connected successfully.

RobertMe replied:
Behavior is the same as current master: https://github.com/mzanetti/kodimote/blob/21b60c0544702498acbe1942003ed5cfe25fece5/libkodimote/kodiconnection.cpp#L407

This code is only ran when:
1. The socket is opened/connected (as it is called from slotConnected)
2. The HTTP calls work (as the version command runs over HTTP)
So this is already done when you are successfully connected.

mzanetti commented:
There are some changes in the sailfish package files which I think are duplicated with another branch

RobertMe commented:
Yep. Thats what the note is about. Without those I couldn't build.

Will rebase when they can build on master (+ fixing above + gitorious issues)

RobertMe commented:
Rebased. Addressed issues of https://gitorious.org/xbmcremote/xbmcremote/merge_requests/38

It's now independent of #1 (and thus can be merged), but Sailfish app isn't directly runnable from the IDE (needs either #1 or a change in the sailfish.pro file (changing TARGET back to harbour-xbmcremote)).

mzanetti commented:
crashes when switching between connections

RobertMe commented:
Could you please try with #15 applied? I would guess it's related to that (as the playlists get reloaded when reconnecting)
